### PR TITLE
Improve parsing

### PR DESCRIPTION
**Pull request by FloopCZ**
- Improve from/to string conversions using `boost::lexical_cast`.
- Improve and speedup `dataframe` accessors.
- Various minor improvements.

### CODE REVIEW

File: include/cxtream/core/utility/string.hpp [hunk 1]
```diff
@@ -23,62 +25,88 @@ namespace cxtream::utility {
...
 }
 
+/// Specialization of string_to() for bool.
+///
+/// "true" and "1" are interpreted as true, "false" and "0" as false.
+/// \throws std::ios_base::failure If an unrecognizable string is provided.
+template<>
+bool string_to<bool>(const std::string& str)
+{
+    if (str == "false" || str == "0") return false;
```
Comment (blazekadam) [inline include/cxtream/core/utility/string.hpp]: I am deeply concerned that we do not support `False`, `Off` and `OFF` here.

File: include/cxtream/core/utility/string.hpp [hunk 2]
```diff
@@ -23,62 +25,88 @@ namespace cxtream::utility {
...
+    if (str == "true" || str == "1") return true;
```
Comment (blazekadam) [inline include/cxtream/core/utility/string.hpp]: Ditto.

File: include/cxtream/core/utility/string.hpp [hunk 3]
```diff
@@ -23,62 +25,88 @@ namespace cxtream::utility {
...
+/// Specialization of to_string() for std::string.
+std::string to_string(const std::string& str)
 {
-    return b ? "true" : "false";
+    return str;
 }
 
-using std::to_string;
+/// Specialization of to_string() for const char *.
+std::string to_string(const char* const& str)
```
Comment (blazekadam) [inline include/cxtream/core/utility/string.hpp]: :+1:

File: include/cxtream/core/utility/string.hpp [hunk 4]
```diff
@@ -23,62 +25,88 @@ namespace cxtream::utility {
...
+{
+    return str;
```
Comment (blazekadam) [inline include/cxtream/core/utility/string.hpp]: :+1: :+1:

File: include/cxtream/core/utility/string.hpp [hunk 5]
```diff
@@ -23,62 +25,88 @@ namespace cxtream::utility {
...
-std::string trim(const std::string& str)
+/// The generated string is either "true" or "false".
+std::string to_string(const bool& b)
 {
-    auto isspace = [](char c) { return std::isspace(c, std::locale::classic()); };
-    auto begin = std::find_if_not(str.begin(), str.end(), isspace);
-    auto end = std::find_if_not(str.rbegin(), str.rend(), isspace).base();
-    if (begin < end) return std::string(begin, end);
-    return std::string{};
+    return b ? "true" : "false";
```
Comment (blazekadam) [inline include/cxtream/core/utility/string.hpp]: Please consider `OFF` and `ON` here. Just consider though :)

File: test/core/utility/string.cpp [hunk 6]
```diff
@@ -30,18 +31,35 @@ BOOST_AUTO_TEST_CASE(test_string_to__string)
...
 }
 
-BOOST_AUTO_TEST_CASE(test_string_to__float_exc)
+BOOST_AUTO_TEST_CASE(test_string_to__bool)
 {
-    std::string str = "0,25";
-    BOOST_CHECK_THROW(string_to<float>(str), std::ios_base::failure);
+    auto b = string_to<bool>("false");
+    static_assert(std::is_same<bool, decltype(b)>{});
+    BOOST_TEST(string_to<bool>("true") == true);
```
Comment (blazekadam) [inline test/core/utility/string.cpp]: Do not forget to test the upper case options as well.

File: test/core/utility/string.cpp [hunk 7]
```diff
@@ -62,14 +89,10 @@ BOOST_AUTO_TEST_CASE(test_float__to_string)
...
-    BOOST_TEST(trim(" hello") == "hello");
-    BOOST_TEST(trim("hello ") == "hello");
-    BOOST_TEST(trim(" hello ") == "hello");
-    BOOST_TEST(trim("\t\n hello\n\t ") == "hello");
-    BOOST_TEST(trim("\t\t\n\n   hello \t\t\n\n  ") == "hello");
+    auto str = to_string(true);
+    static_assert(std::is_same<std::string, decltype(str)>{});
+    BOOST_TEST(to_string(true) == "true");
+    BOOST_TEST(to_string(false) == "false");
 }
```
Comment (blazekadam) [inline test/core/utility/string.cpp]: I suppose you failed to test string to string `const char* const& str` option! :-1:

**Comment by FloopCZ** [comment]
Brutal review revenge. :-D

File: include/cxtream/core/utility/string.hpp (hunk 5 above)
Comment (FloopCZ) [reply include/cxtream/core/utility/string.hpp]: Considered. Denied. :eyeglasses:

File: test/core/utility/string.cpp (hunk 7 above)
Comment (FloopCZ) [reply test/core/utility/string.cpp]: Done.

**Comment by FloopCZ** [comment]
Thank you for the review, I have addressed all the issues.

**Comment by FloopCZ** [comment]
@blazekadam Oh man... Effort... :-D

**Pull request merged**